sound/soc/codecs/rt712-sdca.c

```diff
@@ -1031,6 +1432,10 @@ static int rt712_sdca_pcm_hw_params(struct snd_pcm_substream *substream,
 	if (!rt712->slave)
 		return -EINVAL;
 
+	/* VA doesn't support AIF3 */
+	if (dai->id == RT712_AIF3 && rt712->version_id == RT712_VA)
+		return -EINVAL;
```

Comment:
in what cases would it be necessary to the AIF3?

I am really concerned that the machine driver would need to know about VA/VB.

It's really problematic if the same device has different capabilities over time. It's no longer the same device IMHO and the partID should have been modified.

Reply:
AIF3 is the DMIC function. The VB integrates the DMIC function in one soundwire interface which supports the multi-lanes. The customers could use the DMIC of RT712 if they don't want another DMIC codec.

> It's really problematic if the same device has different capabilities over time. It's no longer the same device IMHO and the partID should have been modified.

Hmm...I'm sorry for that. I can't decide the partID if the codec will be modified.